ReSharper.FSharp/test/data/parsing/Expr - CompExpr - LetUseAndBang 01.fs

```diff
@@ -0,0 +1,6 @@
+async {
+     let! x = ()
+     and! y = ()
+     use! z = ()
+     ()
```

Comment:
I think the `use!` part doesn't cover any special logic here.

Reply:
Wel,, to be honest this test just covers the fact that use! token is processed at all. But it almost looks the same as SynExpr LetAnd 01. I can delete this one if you want or that one if you want to keep use! token in tests.

Reply:
It'd be better to have a separate test for `use!` if you want to test it. In general, we try to make each parser test as simple as possible.

Reply:
I think I've found a suspicious thing that this tests actually shows:
Please look at gold file. I expect there would be 2 LET_OR_USE_EXPR with same level of nesting, e.g. 
"let! .. and! .." 
"use! .."

But in fact "use! .." expression is a child of "let! .. and! .."
So it's like 
"let! .. and! .." 
    "use! .."

This seems like a thing you've talked about - when LetOrUseBang is reported from fcs, first "let!" expression range is actually "let! .. and! .. use!.." instead of just "let! .. and! .."
Second LetOrUseBang expression is okay - it has a proper range of "use! .."

Reply:
> But in fact "use! .." expression is a child of "let! .. and! .."

It looks correct to me. The `use!` expression is the `in` expressions in the `let!` one here. It's the same for `let` expressions, e.g.:

```fsharp
```

Reply:
It's a different story for top-level bindings as they don't have the obligatory `in` expression. When bindings are placed inside expressions they always have this recursive structure.

Reply:
Or maybe this shows nothing because it's like this with every LetOrUse/LetOrUseBang - they are nested. Can you clarify this is by design? I will remove this test then as you right and it actually cover nothing.

Reply:
Yes, it'd by design indeed. 🙂

Prior to F# introduced the indentation-based syntax it was obligatory to write explicit `in` in such nested bindings, and it easier understand this structure if add it:
```fsharp
do
    let a = 1 in
    let b = 2 in
    ()
```